### DISCUSSION TITLE
Is there a plugin or setup to include git history of a page in said page?

### DISCUSSION THREAD

**Discussion by Andre601** [mkdocs/mkdocs#3524] [Q&A]
Wondering if there is any plugin, extension or other setup to include a file's git history, meaning all the changes it actually received.

I work on a wiki. And in true Wikipedia-like fashion would I like to expose a page's edit history using the git commit history as its base.

I was looking at the `git` attribute that the MkDocs Macros plugin - which I also use - is providing, but that only gives a more general collection of info, such as the actual last commit of the git repository in general and not a particular file... Also, no way of obtaining the commit history itself.

So, is there something already?

**Comment by redtide** [comment]
I tried to make a search, there was one called git timeline plugin but the repository was deleted.
I guess it could be something that can be done in python to run git commands locally or in CI and produce a result in some json file to parse in mkdocs?

**Reply by redtide** [reply]
I gave a look again but I wasn't able to find something more advanced that what is also listed in mkdocs [catalog' git section](https://github.com/mkdocs/catalog#-git-repos--info).
I believe something that advanced would be good to have as hook using a python git library; I see some plugins depends on GitPython.
